### PR TITLE
Added week 5

### PR DESCRIPTION
**Pull request by shreythecray**
Updated steps 1-3 in week 5:
- create a flowchart
- write CFP
- start developing

### CODE REVIEW

**Review by emsesc — CHANGES_REQUESTED** [review]
We need to reformat the files (content looks great) with the new checkboxes instead of "to move on..."

Reference the week 1 response files for a good example

File: .bit/responses/5.1-Week 5 Step 1.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+## Laying out the groundwork
+
+### ✅  Task:
+
+- [ ]  Create a new [figma](figma.com) file named "Serverless Camp Project"
+- [ ]  In this file, create a flowchart showing the flow of information and components involved in your project including any Azure Functions and APIs you plan on using
+- [ ]  Export the flowchart as a png file 
+- [ ]  Create a branch named `week5`
+- [ ]  Create a folder in this repo where you will be saving all of your project related files, called "project"
+- [ ]  Upload the flowchart into the project folder and commit your changes to move on
```
Comment (lazyplatypus) [inline .bit/responses/5.1-Week 5 Step 1.md:20]: combine 19/20. creating a folder should not be a step

File: .bit/responses/5.1-Week 5 Step 1.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+name: Week 5 Step 1
+---
+Week 5 Step 1 ⬤◯◯◯ | 🕐 Estimated completion: 30 minutes
+
+## Laying out the groundwork
+
+### ✅  Task:
+
+- [ ]  Create a new [figma](figma.com) file named "Serverless Camp Project"
+- [ ]  In this file, create a flowchart showing the flow of information and components involved in your project including any Azure Functions and APIs you plan on using
```
Comment (lazyplatypus) [inline .bit/responses/5.1-Week 5 Step 1.md:16]: combnie 15 + 16

File: .bit/responses/5.1-Week 5 Step 1.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+- [ ]  Export the flowchart as a png file 
+- [ ]  Create a branch named `week5`
+- [ ]  Create a folder in this repo where you will be saving all of your project related files, called "project"
+- [ ]  Upload the flowchart into the project folder and commit your changes to move on
+
+### Mapping it out
+
```
Comment (lazyplatypus) [inline .bit/responses/5.1-Week 5 Step 1.md:23]: why is this 3rd person. stick to one POV

File: .bit/responses/5.1-Week 5 Step 1.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+- Document your project and communicate how it works to others (this is really valuable when someone else is helping you debug your code)
+- Prevent major technical issues with your code before you event start
+- Ensure that you are more effectively using all the technologies available to you
+- You will need to refer back to the flowchart while developing to review your work and sometimes come up with new ideas
+
+<details>
+<summary>:exclamation: How do I create a flowchart?</summary>
+</br>
+Part A: Purpose
+1. Create a checklist of all of the components in your project you want to use, including APIs, functions, databases, packages/libraries, etc.
```
Comment (lazyplatypus) [inline .bit/responses/5.1-Week 5 Step 1.md:38]: this should not be in the dropdown. this information is necessary to complete the assignment

File: .bit/responses/5.1-Week 5 Step 1.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+3. Below each shape, create bullet points on what purpose they serve in your project (be as specific as you want to)
+
+Part B: Interactions
+1. Create a new circular shape/block for each of the components and write their names inside each shape
+2. Draw lines on all of the technologies that interact with each other
+3. Near each line, use the previous bullet points you wrote to describe the relationship between the technologies starting with action verbs (e.g. between JS code/front-end --> Azure Function "Sends an HTTP Post Request with uploaded image)
+
+Part C: Review
+1. Get a friend/mentor to look over the flowchart and try to explain back the program to you
```
Comment (lazyplatypus) [inline .bit/responses/5.1-Week 5 Step 1.md:47]: This should be in the checkbox ( maybe another step

**Review by lazyplatypus — CHANGES_REQUESTED** [review]
change plz

File: .bit/responses/5.1-Week 5 Step 1.md (hunk 5 above)
Comment (shreythecray) [reply .bit/responses/5.1-Week 5 Step 1.md:47]: Did you mean checkbox for 1. 2. 3. or for each Part?

Comment (lazyplatypus) [reply .bit/responses/5.1-Week 5 Step 1.md:47]: each part sohuld be another step

Comment (lazyplatypus) [reply .bit/responses/5.1-Week 5 Step 1.md:47]: you need to create week 5 step 2, 3

File: .bit/responses/5.3-Week 5 Step 3.md [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+There are 3 main things to focus on when you are getting feedback on and improving your flowchart:
+1. Logic:
+  - Does the flowchart make sense?
+  - Do the arrow directions makes sense?
+  - Are the descriptions for the interactions between technologies logically correct?
+2. Accuracy: 
+  - Did you include all components required for this project?
+  - Do the arrows represent accurate relationships?
+  - Are the descriptions accurate to what purposes the components serve?
+3. Readaibility and visual appeal:
```
Comment (emsesc) [inline .bit/responses/5.3-Week 5 Step 3.md:31]: Small spelling error

File: .bit/responses/5.4-Week 5 Step 4.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+---
+Week 5 Step 4 ◯◯◯⬤◯◯ | 🕐 Estimated completion: 25 minutes
+
+## Timeline
+
+### ✅  Task:
+
+- [ ] Develop a draft of your timeline in a new file called "timeline.md" in the projects folder
+- [ ] Ask a mentor to review your gameplan
+- [ ] Once your timeline is mentor approved, convert your timeline into issues in this repo
```
[the remark this answers is not in the document]
Comment (emsesc) [reply .bit/responses/5.4-Week 5 Step 4.md:17]: @lazyplatypus what do you think about asking them to create a separate repo for their final project?

File: .bit/responses/5.4-Week 5 Step 4.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+Sometimes you might become lucky and end up having extra time on your hands. How do you plan on maximizing the impact of this time so you can ship the best version of your project at the end?
+
+Be clearn on how to manage your own time and hold yourself accountable! Ask your peers and friends for help if you need it.
+
+### Getting feedback
+
+Your mentor knows best! They are professionals who have been working in the tech industry on projects much more complex and that have covered all sorts of timelines and team sizes. Get their feedback on your plan and see if they have any recommendations on how you can build your project more efficiently.
+
+### Converting tasks into issues
```
Comment (emsesc) [inline .bit/responses/5.4-Week 5 Step 4.md:51]: We need examples for each of these (maybe templates?) - that way it seems less overwhelming

File: .bit/responses/5.6-Week 5 Step 6.md [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+step: 6
+name: Week 5 Step 6
+---
+Week 5 Step 6 ◯◯◯◯◯⬤ | 🕐 Estimated completion: 25 minutes
+
+## Documentation
+
+### ✅  Task:
+
+- [ ] Document your first component in blog.md
```
Comment (emsesc) [inline .bit/responses/5.6-Week 5 Step 6.md:15]: Specify the "component" specifically

File: .bit/responses/5.6-Week 5 Step 6.md [hunk 10]
```diff
@@ -0,0 +1,32 @@
...
+* Break down your code:
+  * Copy all of your code and paste it into a code snippet in your blog
+  * Break it down into multiple snippets in order of how you wrote the code
+  * Before each snippet, write a description of what the code is for and how it works
+  * Define keywords that were new to you or would be to your audience
+  * Tip: use substeps to make your blog easier to follow
+
+If you have completed multiple components of your project this week, document them as you go.
+
+## **:camping: To move on, commit your code in a branch called "week5", update your blog, and comment a summary of what you have developed so far!**
```
Comment (emsesc) [inline .bit/responses/5.6-Week 5 Step 6.md:32]: We've removed the "to move on" section, all of this will go in the checklist at the top

File: .bit/responses/5.6-Week 5 Step 6.md [hunk 11]
```diff
@@ -0,0 +1,32 @@
...
+## Documentation
+
+### ✅  Task:
+
+- [ ] Document your first component in blog.md
+- [ ] Commit your changes in the `week5` branch
+
+Once you have successfully completed developing your first component, you can start documenting your project blog. Remember that the purpose of this blog is to help someone else create your project from scratch, so as you start writing, you will need to break down your code as much as possible and explain it so that someone else can learn from your example.
+
+### Quick tips
```
Comment (emsesc) [inline .bit/responses/5.6-Week 5 Step 6.md:20]: Once again, I think solid examples would be super helpful (you could pull some from blogs that have already been written, or attach screenshots etc.)

File: .bit/responses/5.5-Week 5 Step 5.md [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+step: 5
+name: Week 5 Step 5
+---
+Week 5 Step 5 ◯◯◯◯⬤◯ | 🕐 Estimated completion: 1-5 hours
+
+## Start Coding!
+
+### ✅  Task:
+
+- [ ] Based on your schedule, begin your first task
```
Comment (emsesc) [inline .bit/responses/5.5-Week 5 Step 5.md:15]: Depending on whether we choose to have them put their project in a second repo or not, these tasks will change

File: .bit/responses/5.5-Week 5 Step 5.md [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+### Developing the first component
+
+Let's look back at the advice we got from the previous step: "start small then build big." Your goal is to get something working - not to developing your final project in 1 week.
+* Begin by just having one working component of the project. Ex:
+  * Press a button on your webpage and receive information
+  * Create a working Azure Function that can return information
+* After that's complete and tested, begin building onto that with more and more features!
+
+### Committing code
+
```
Comment (emsesc) [inline .bit/responses/5.5-Week 5 Step 5.md:28]: Also add here that they should **name** commits correctly so that they are easy to search for

**Review by emsesc — CHANGES_REQUESTED** [review]
Great start on the content side of things! Here are a couple key things:

- [x] Formatting
* The circles
* putting things in drop downs
* Emojis?
* IssueComment (check your first few issues)

- [ ] Content
* The sections are a bit wordy (try condensing them)
* Less describing and more showing (give examples from previous projects/blogs instead of describing what they should look like)
* Maybe provide templates? like a template for creating issues

@lazyplatypus feedback on if they should create a separate repository for their final project? I would argue to do that, since files can get really messay (also, we have a package.json file in this repo so it will likely interfere with theirs)

**Comment by lazyplatypus** [comment]
I would highly prefer to do it in a separate folder within the same repo. it helsp with consistency. @emsesc what do you think?

File: .bit/responses/5.1-Week 5 Step 1.md [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+
+### Researching APIs (and other technologies)
+
+There are thousands of amazing APIs that all have various functions, and there is no limit to the number of APIs you can use for your project. Here are a couple ways you can find the right API for your project:
+
+- If you already know what purpose you want your API to serve (e.g. find a location, send an SMS, ...) start with a simple Google search or filter through online lists such as [rapidapi.com](rapidapi.com)
+- If you're looking for project inspiration through cool new APIs, online lists such as [rapidapi.com](https://rapidapi.com/categories) have tags/categories that you can filter through to get some ideas 
+- When in doubt, talk to people in the tech industry or try watching some API workshops to see what is out there
+
+Some of the most popular API categories previously seen in Serverless Camps include:
```
[the remark this answers is not in the document]
Comment (shreythecray) [reply .bit/responses/5.1-Week 5 Step 1.md]: Need to add Evelyn's project to the website

File: .bit/responses/5.6-Week 5 Step 6.md (hunk 9 above)
Comment (shreythecray) [reply .bit/responses/5.6-Week 5 Step 6.md:15]: Hey @emsesc can you clarify what you mean by this?

Comment (emsesc) [reply .bit/responses/5.6-Week 5 Step 6.md:15]: @shreythecray Yup! I meant define what component you're talking about. Essentially, what is the component you're referring to?

Comment (shreythecray) [reply .bit/responses/5.6-Week 5 Step 6.md:15]: I'm not sure if I can do this here since they will be deciding on their own which component they will be working on

**Review by emsesc — APPROVED** [review]

**Review by lazyplatypus — APPROVED** [review]
